homeassistant/components/alarm_control_panel/canary.py

```diff
@@ -0,0 +1,83 @@
+import logging
+
+from homeassistant.components.alarm_control_panel import AlarmControlPanel
+from homeassistant.components.canary import DATA_CANARY
+from homeassistant.const import STATE_UNKNOWN, STATE_ALARM_DISARMED, \
+    STATE_ALARM_ARMED_AWAY, STATE_ALARM_ARMED_NIGHT
+
+DEPENDENCIES = ['canary']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Canary alarms."""
+    data = hass.data[DATA_CANARY]
+    devices = []
+
+    for location in data.locations:
+        devices.append(CanaryAlarm(data, location.location_id))
+
+    add_devices(devices, True)
+
+
+class CanaryAlarm(AlarmControlPanel):
+    """Representation of a Canary alarm control panel."""
+
+    def __init__(self, data, location_id):
+        """Initialize a Canary security camera."""
+        super().__init__()
+        self._data = data
+        self._location_id = location_id
+
+    @property
+    def name(self):
+        """Return the name of the alarm."""
+        location = self._data.get_location(self._location_id)
+        return location.name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        from canary.api import LOCATION_MODE_AWAY, LOCATION_MODE_HOME, \
+            LOCATION_MODE_NIGHT
+
+        location = self._data.get_location(self._location_id)
+
+        mode = location.mode
+        if mode.name == LOCATION_MODE_AWAY:
+            return STATE_ALARM_ARMED_AWAY
+        elif mode.name == LOCATION_MODE_HOME:
+            return STATE_ALARM_DISARMED
+        elif mode.name == LOCATION_MODE_NIGHT:
+            return STATE_ALARM_ARMED_NIGHT
+        else:
+            return STATE_UNKNOWN
+
+    def alarm_disarm(self, code=None):
+        """Send disarm command."""
+        from canary.api import LOCATION_MODE_HOME
+        self._data.set_location_mode(self._location_id, LOCATION_MODE_HOME)
+
+    def alarm_arm_home(self, code=None):
+        """Send arm home command."""
+        from canary.api import LOCATION_MODE_HOME
+        self._data.set_location_mode(self._location_id, LOCATION_MODE_HOME)
```

Comment:
Above you disarm with the same command. How does this work?

Reply:
For Canary there are 3 location mode, home, away and night. 

Setting mode to home is essentially the same as disarm

Maybe I should map arm_home to Canary location mode away? What do you think?

Reply:
If you only have those three modes, I think you shouldn't implement `alarm_arm_home` at all.

Reply:
But then on the UI, the "Arm Home" button is still visible and clickable, if I don't implement it, what happens if user click on it?

Reply:
Yeah ok, you should implement it but just log a warning.

We haven't implemented supported features for alarm control panel yet. If we do the frontend could be tailored to show only the services implemented by each platform.

Reply:
I made change to how the mapping is done:

HASS.ARMED_HOME => Canary Home Mode
HASS.ARMED_AWAY => Canary Away Mode
HASS.ARMED_NIGHT => Canary Night Mode
HASS.DISARM => Canary Privacy Mode (This turns off recording essentially acting as disarmed)